# Who eats the Food we Grow?

- Food: refers to the total amount of the food item available as human food during the reference period.

- Feed: refers to the quantity of the food item available for feeding to the livestock and poultry during the reference period.

El dataset es demasiado grande, con 21477 filas, y 63 columnas. Asi que voy simplificar mi analisis, al extraer un subconjunto del dataset: solo para Norteamerica, y solo la decada de los 2000's

In [33]:
import pandas as pd
import numpy as np

df_food = pd.read_csv('../Data_Base/FAO.csv', sep=',', encoding='ISO-8859-1')
df_food.shape

(21477, 63)

In [34]:
df_food.head(2)

Area Abbreviation  Area Code         Area  Item Code  \
0               AFG          2  Afghanistan       2511   
1               AFG          2  Afghanistan       2805   

                       Item  Element Code Element         Unit  latitude  \
0        Wheat and products          5142    Food  1000 tonnes     33.94   
1  Rice (Milled Equivalent)          5142    Food  1000 tonnes     33.94   

   longitude  ...   Y2004   Y2005   Y2006   Y2007   Y2008   Y2009   Y2010  \
0      67.71  ...  3249.0  3486.0  3704.0  4164.0  4252.0  4538.0  4605.0   
1      67.71  ...   419.0   445.0   546.0   455.0   490.0   415.0   442.0   

    Y2011  Y2012  Y2013  
0  4711.0   4810   4895  
1   476.0    425    422  

[2 rows x 63 columns]

Usando Filtrado por condiciones, para obtener un Data Frame de solo los paises de NorteAmerica

In [36]:
df_food = df_food[(df_food.Area == 'Mexico') | (df_food.Area == 'Canada') | (df_food.Area == 'United States of America')]

In [37]:
df_food.groupby('Area').count()

Area Abbreviation  Area Code  Item Code  Item  \
Area                                                                      
Canada                                  140        140        140   140   
Mexico                                  133        133        133   133   
United States of America                141        141        141   141   

                          Element Code  Element  Unit  latitude  longitude  \
Area                                                                         
Canada                             140      140   140       140        140   
Mexico                             133      133   133       133        133   
United States of America           141      141   141       141        141   

                          Y1961  ...  Y2004  Y2005  Y2006  Y2007  Y2008  \
Area                             ...                                      
Canada                      140  ...    140    140    140    140    140   
Mexico                      133  ...    133    133    133    133    133   
United States of America    141  ...    141    141    141    141    141   

                          Y2009  Y2010  Y2011  Y2012  Y2013  
Area                                                         
Canada                      140    140    140    140    140  
Mexico                      133    133    133    133    133  
United States of America    141    141    141    141    141  

[3 rows x 62 columns]

In [38]:
df_food = df_food.drop(df_food.iloc[:,10:49], axis=1)
df_food = df_food.drop(df_food.iloc[:, -4:], axis=1)
df_food.shape

(414, 20)

Finalmente, se obtiene un dataset solo con la informacion relevante, del año 2000 al 2009, y paises de America del Norte

In [39]:
df_food.head(2)

Area Abbreviation  Area Code    Area  Item Code                Item  \
3386               CAN         33  Canada       2511  Wheat and products   
3387               CAN         33  Canada       2511  Wheat and products   

      Element Code Element         Unit  latitude  longitude   Y2000   Y2001  \
3386          5521    Feed  1000 tonnes     56.13    -106.35  3447.0  3342.0   
3387          5142    Food  1000 tonnes     56.13    -106.35  2724.0  2742.0   

       Y2002   Y2003   Y2004   Y2005   Y2006   Y2007   Y2008   Y2009  
3386  4066.0  3442.0  5095.0  4152.0  4150.0  2199.0  2197.0  2629.0  
3387  2728.0  2740.0  2815.0  2827.0  2854.0  2908.0  2897.0  2895.0

Agregando una Columna que contiene la suma de las columnas desde el Y2000 al Y2009

In [43]:
df_food['Suma'] = df_food.iloc[:, 10:].sum(axis=1)
df_food.head(2)

Area Abbreviation  Area Code    Area  Item Code                Item  \
3386               CAN         33  Canada       2511  Wheat and products   
3387               CAN         33  Canada       2511  Wheat and products   

      Element Code Element         Unit  latitude  longitude  ...   Y2001  \
3386          5521    Feed  1000 tonnes     56.13    -106.35  ...  3342.0   
3387          5142    Food  1000 tonnes     56.13    -106.35  ...  2742.0   

       Y2002   Y2003   Y2004   Y2005   Y2006   Y2007   Y2008   Y2009     Suma  
3386  4066.0  3442.0  5095.0  4152.0  4150.0  2199.0  2197.0  2629.0  69438.0  
3387  2728.0  2740.0  2815.0  2827.0  2854.0  2908.0  2897.0  2895.0  56260.0  

[2 rows x 21 columns]

Tener Cuidado con esta columna *Suma*, porque como demostraremos al hacer un simple agrupamiento, no coincide si se suman todos los años desde 2000 al 2009

In [44]:
df_food.groupby('Element').sum()

Area Code  Item Code  Element Code  latitude  longitude     Y2000  \
Element                                                                      
Feed         12888     256471        524495   3834.41   -9625.21  437378.0   
Food         42657     857540       1640298  12396.27  -32398.05  815608.0   

            Y2001     Y2002     Y2003     Y2004     Y2005     Y2006     Y2007  \
Element                                                                         
Feed     434486.0  406065.0  436204.0  455094.0  442156.0  405680.0  419658.0   
Food     814777.0  830405.0  837719.0  851118.0  845164.0  850922.0  862750.0   

            Y2008     Y2009        Suma  
Element                                  
Feed     402859.0  379783.0   8438726.0  
Food     853438.0  854852.0  16833506.0

In [45]:
df_aux = df_food.groupby('Element').sum()

Los valores obtenidos a continuacion son diferentes a los de la Columna *Suma*. Obtengo el mismo resultado con Excel, lo que indica esta bien.

In [52]:
df_aux.iloc[:,-11:-1].sum(axis=1)

Element
Feed    4219363.0
Food    8416753.0
dtype: float64

Mejor borremos la columana *Suma* y cada vez que necesite calcular las uma por alguna razon, la evaluo ahi mismo

In [53]:
df_food.drop('Suma', axis=1, inplace=True)

In [54]:
df_food.head(2)

Area Abbreviation  Area Code    Area  Item Code                Item  \
3386               CAN         33  Canada       2511  Wheat and products   
3387               CAN         33  Canada       2511  Wheat and products   

      Element Code Element         Unit  latitude  longitude   Y2000   Y2001  \
3386          5521    Feed  1000 tonnes     56.13    -106.35  3447.0  3342.0   
3387          5142    Food  1000 tonnes     56.13    -106.35  2724.0  2742.0   

       Y2002   Y2003   Y2004   Y2005   Y2006   Y2007   Y2008   Y2009  
3386  4066.0  3442.0  5095.0  4152.0  4150.0  2199.0  2197.0  2629.0  
3387  2728.0  2740.0  2815.0  2827.0  2854.0  2908.0  2897.0  2895.0

Realizo un agrupamiento por Pais: por ejemplo, Canada tiene 140 registros, y para el año 2000, habia disponible 801 T de comida tanto para consumo humano como para alimentar animales. 

In [63]:
df_food.groupby('Area').agg(['count', 'mean']).iloc[:,10:]

/tmp/ipykernel_2146/1386694542.py:1: FutureWarning: ['Area Abbreviation', 'Item', 'Element', 'Unit'] did not aggregate successfully. If any error is raised this will raise in a future version of pandas. Drop these columns/ops to avoid this warning.
  df_food.groupby('Area').agg(['count', 'mean']).iloc[:,10:]


Y2000              Y2001              Y2002  \
                         count         mean count         mean count   
Area                                                                   
Canada                     140   801.757143   140   790.671429   140   
Mexico                     133  1321.631579   133  1408.766917   133   
United States of America   141  6843.709220   141  6746.120567   141   

                                      Y2003              Y2004               \
                                 mean count         mean count         mean   
Area                                                                          
Canada                     794.007143   140   840.914286   140   813.950000   
Mexico                    1408.548872   133  1454.781955   133  1464.390977   
United States of America  6652.283688   141  6827.723404   141  7074.432624   

                         Y2005              Y2006              Y2007  \
                         count         mean count         mean count   
Area                                                                   
Canada                     140   823.321429   140   792.578571   140   
Mexico                     133  1428.052632   133  1479.323308   133   
United States of America   141  6965.418440   141  6729.723404   141   

                                      Y2008              Y2009               
                                 mean count         mean count         mean  
Area                                                                         
Canada                     834.964286   140   816.028571   140   763.478571  
Mexico                    1504.127820   133  1548.556391   133  1489.646617  
United States of America  6847.262411   141  6638.971631   141  6593.085106

Mediante una funcion de agrupamiento doble, logro ilustrar que tanta candidad de comida estaba disponible por pais, y por elemento: para consumo animal o humano. Se reduce la carga cognitiva, y creo la table es muy clara

In [73]:
df_food.groupby(['Area', 'Element']).sum().iloc[:,5:]

Y2000     Y2001     Y2002     Y2003  \
Area                     Element                                           
Canada                   Feed      49803.0   48126.0   47437.0   55045.0   
                         Food      62443.0   62568.0   63724.0   62683.0   
Mexico                   Feed      36194.0   41133.0   40154.0   43558.0   
                         Food     139583.0  146233.0  147183.0  149928.0   
United States of America Feed     351381.0  345227.0  318474.0  337601.0   
                         Food     613582.0  605976.0  619498.0  625108.0   

                                     Y2004     Y2005     Y2006     Y2007  \
Area                     Element                                           
Canada                   Feed      49719.0   50374.0   45316.0   49737.0   
                         Food      64234.0   64891.0   65645.0   67158.0   
Mexico                   Feed      42871.0   38100.0   41787.0   41411.0   
                         Food     151893.0  151831.0  154963.0  158638.0   
United States of America Feed     362504.0  353682.0  318577.0  328510.0   
                         Food     634991.0  628442.0  630314.0  636954.0   

                                     Y2008     Y2009  
Area                     Element                      
Canada                   Feed      48320.0   40602.0  
                         Food      65924.0   66285.0  
Mexico                   Feed      46366.0   39782.0  
                         Food     159592.0  158341.0  
United States of America Feed     308173.0  299399.0  
                         Food     627922.0  630226.0